Connectors/CSi/Speckle.Connectors.CSiShared/Bindings/CsiSharedBasicConnectorBinding.cs

```diff
@@ -45,11 +53,11 @@ ITopLevelExceptionHandler topLevelExceptionHandler
 
   public DocumentModelStore GetDocumentState() => _store;
 
-  public void AddModel(ModelCard model) => _store.AddModel(model);
+  public void AddModel(ModelCard model) => _threadContext.RunOnMain(() => _store.AddModel(model));
 
-  public void UpdateModel(ModelCard model) => _store.UpdateModel(model);
+  public void UpdateModel(ModelCard model) => _threadContext.RunOnMain(() => _store.UpdateModel(model));
```

Comment:
This is an unawaited Task now. Need to use an exception handler

Reply:
see updated commit